src/test/java/uk/gov/pay/adminusers/service/ServiceUpdaterTest.java

```diff
@@ -189,4 +193,27 @@ public void shouldError_IfAGatewayAccountAlreadyAssignedToAService() throws Exce
         verify(serviceDao, times(0)).merge(serviceEntity);
     }
 
+    @Test
+    public void shouldUpdateServiceNameSuccessfully() {
+        String serviceId = randomUuid();
+        String nameToUpdate = "new-cy-name";
+        ServiceUpdateRequest request = ServiceUpdateRequest.from(new ObjectNode(JsonNodeFactory.instance, ImmutableMap.of(
+                "path", new TextNode("service_name"),
+                "value", new ObjectNode(JsonNodeFactory.instance, ImmutableMap.of("cy", new TextNode(nameToUpdate))),
+                "op", new TextNode("replace"))));
+        ServiceEntity serviceEntity = mock(ServiceEntity.class);
+
+        when(serviceDao.findByExternalId(serviceId)).thenReturn(Optional.of(serviceEntity));
+        when(serviceEntity.toService()).thenReturn(Service.from());
+
+        InOrder inOrder = Mockito.inOrder(serviceDao);
+
+        Optional<Service> maybeService = updater.doUpdate(serviceId, request);
+
+        assertThat(maybeService.isPresent(), is(true));
+        ServiceNameEntity serviceNameEntity = ServiceNameEntity.from(SupportedLanguage.WELSH, nameToUpdate);
+        inOrder.verify(serviceDao, times(1)).findByExternalId(serviceId);
+        inOrder.verify(serviceDao, times(1)).merge(serviceEntity);
+        verify(serviceEntity, times(1)).addOrUpdateServiceName(serviceNameEntity);
```

Comment:
Ah, I think I wasn’t clear. I meant to do something like this:

```java
InOrder inOrder = Mockito.inOrder(serviceEntity, serviceDao);
// …
inOrder.verify(serviceEntity).addOrUpdateServiceName(serviceNameEntity);
inOrder.verify(serviceDao).merge(serviceEntity);
```

We want to ensure the `serviceEntity` is updated before it’s merged via the `serviceDao`.

Also, `times(1)` isn’t needed: it’s the default.